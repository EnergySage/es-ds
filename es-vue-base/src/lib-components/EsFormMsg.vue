<template>
    <b-alert
        v-if="message"
        :show="dismissCountDown"
        :variant="variant"
        dismissible
        class="form-msg"
        fade
        @dismissed="dismissCountDown=0"
        @dismiss-count-down="countDownChanged">
        <div class="d-flex">
            <div class="pr-2">
                <IconCircleAlert v-if="variant === 'danger'" />
                <IconCircleCheck v-if="variant === 'success'" />
            </div>
            {{ message }}
        </div>
    </b-alert>
</template>
<script>
import {
    BAlert,
} from 'bootstrap-vue';
import IconCircleAlert from '@/src/lib-icons/circle-alert.vue';
import IconCircleCheck from '@/src/lib-icons/circle-check.vue';

export default {
    name: 'EsFormMsg',
    components: {
        BAlert,
        IconCircleAlert,
        IconCircleCheck,
    },
    props: {
        message: {
            type: String,
            default: '',
            required: false,
        },
        timeout: {
            type: Number,
            default: 5,
            required: false,
        },
        variant: {
            type: String,
            default: 'danger',
            required: false,
            validator: (val) => ['danger', 'success'].includes(val),
        },
    },
    data() {
        return {
            dismissCountDown: this.timeout,
        };
    },
    watch: {
        message() {
            this.dismissCountDown = this.timeout;
        },
    },
    methods: {
        countDownChanged(currentCountDown) {
            this.dismissCountDown = currentCountDown;
            if (currentCountDown === 0) {
                this.$emit('hidden');
            }
        },
    },
};
</script>
<style lang="scss" scoped>
@import '~@energysage/es-bs-base/scss/includes';

.form-msg {
  flex: 0 0 100%;
  padding-right: 2.5rem;
}
</style>
