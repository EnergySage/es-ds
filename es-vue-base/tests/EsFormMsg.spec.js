import { mount } from '@vue/test-utils';
import EsFormMsg from '@/src/lib-components/EsFormMsg.vue';
import jestVue from '@/tests/jest.vue.config';

describe('EsFormMsg', () => {
    // Basic test; should exist for most components
    test('<EsFormMsg />', async () => {
        const wrapper = mount(EsFormMsg, {
            ...jestVue,
            propsData: {
                show: true,
            },
        });
        const a11y = await axe(wrapper.element);

        expect(wrapper.vm).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();
        expect(a11y).toHaveNoViolations();
    });
});
