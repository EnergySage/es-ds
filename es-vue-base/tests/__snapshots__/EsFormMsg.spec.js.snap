// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EsFormMsg <EsFormMsg variant="danger" show="false" /> 1`] = `
"<transition-stub name="" css="true" enterclass="" leaveclass="show" entertoclass="show" leavetoclass="" enteractiveclass="fade" leaveactiveclass="fade" class="variant-danger">
  <!---->
</transition-stub>"
`;

exports[`EsFormMsg <EsFormMsg variant="danger" show="true" /> 1`] = `
"<transition-stub name="" css="true" enterclass="" leaveclass="show" entertoclass="show" leavetoclass="" enteractiveclass="fade" leaveactiveclass="fade" class="variant-danger">
  <div role="alert" aria-live="polite" aria-atomic="true" class="alert alert-dismissible alert-danger"><button type="button" aria-label="Close" class="close">×</button>
    <div class="d-flex pr-100">
      <div class="p-25 icon-wrapper icon-color-danger"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" style="height: 24px; width: 24px;">
          <path d="M23 12c0 6.075-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1s11 4.925 11 11Z" fill="none"></path>
          <path fill-rule="evenodd" clip-rule="evenodd" d="M12 3a9 9 0 1 0 0 18 9 9 0 0 0 0-18ZM1 12C1 5.925 5.925 1 12 1s11 4.925 11 11-4.925 11-11 11S1 18.075 1 12Z"></path>
          <path fill-rule="evenodd" clip-rule="evenodd" d="M12 6a1 1 0 0 1 1 1v6a1 1 0 1 1-2 0V7a1 1 0 0 1 1-1Zm0 9.5a1 1 0 0 1 1 1v.5a1 1 0 1 1-2 0v-.5a1 1 0 0 1 1-1Z"></path>
        </svg>
        <!---->
        <!---->
      </div> Main Content
    </div>
  </div>
</transition-stub>"
`;

exports[`EsFormMsg <EsFormMsg variant="primary" show="false" /> 1`] = `
"<transition-stub name="" css="true" enterclass="" leaveclass="show" entertoclass="show" leavetoclass="" enteractiveclass="fade" leaveactiveclass="fade" class="variant-primary">
  <!---->
</transition-stub>"
`;

exports[`EsFormMsg <EsFormMsg variant="primary" show="true" /> 1`] = `
"<transition-stub name="" css="true" enterclass="" leaveclass="show" entertoclass="show" leavetoclass="" enteractiveclass="fade" leaveactiveclass="fade" class="variant-primary">
  <div role="alert" aria-live="polite" aria-atomic="true" class="alert alert-dismissible alert-primary"><button type="button" aria-label="Close" class="close">×</button>
    <div class="d-flex pr-100">
      <div class="p-25 icon-wrapper icon-color-primary">
        <!---->
        <!----> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" style="height: 24px; width: 24px;">
          <path d="M23 12c0 6.075-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1s11 4.925 11 11Z" fill="none"></path>
          <path fill-rule="evenodd" clip-rule="evenodd" d="M12 3a9 9 0 1 0 0 18 9 9 0 0 0 0-18ZM1 12C1 5.925 5.925 1 12 1s11 4.925 11 11-4.925 11-11 11S1 18.075 1 12Z"></path>
          <path fill-rule="evenodd" clip-rule="evenodd" d="M11 7a1 1 0 0 1 1-1h.01a1 1 0 1 1 0 2H12a1 1 0 0 1-1-1Zm1 3a1 1 0 0 1 1 1v5a1 1 0 1 1-2 0v-5a1 1 0 0 1 1-1Z"></path>
        </svg></div> Main Content
    </div>
  </div>
</transition-stub>"
`;

exports[`EsFormMsg <EsFormMsg variant="success" show="false" /> 1`] = `
"<transition-stub name="" css="true" enterclass="" leaveclass="show" entertoclass="show" leavetoclass="" enteractiveclass="fade" leaveactiveclass="fade" class="variant-success">
  <!---->
</transition-stub>"
`;

exports[`EsFormMsg <EsFormMsg variant="success" show="true" /> 1`] = `
"<transition-stub name="" css="true" enterclass="" leaveclass="show" entertoclass="show" leavetoclass="" enteractiveclass="fade" leaveactiveclass="fade" class="variant-success">
  <div role="alert" aria-live="polite" aria-atomic="true" class="alert alert-dismissible alert-success"><button type="button" aria-label="Close" class="close">×</button>
    <div class="d-flex pr-100">
      <div class="p-25 icon-wrapper icon-color-success">
        <!----> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" style="height: 24px; width: 24px;">
          <path d="M23 12c0 6.075-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1s11 4.925 11 11Z" fill="none"></path>
          <path fill-rule="evenodd" clip-rule="evenodd" d="M16.64 8.732a1 1 0 0 1 .128 1.408l-5 6a1 1 0 0 1-1.475.067l-3-3a1 1 0 1 1 1.414-1.414l2.226 2.226 4.299-5.16a1 1 0 0 1 1.408-.127Z"></path>
          <path fill-rule="evenodd" clip-rule="evenodd" d="M12 3a9 9 0 1 0 0 18 9 9 0 0 0 0-18ZM1 12C1 5.925 5.925 1 12 1s11 4.925 11 11-4.925 11-11 11S1 18.075 1 12Z"></path>
        </svg>
        <!---->
      </div> Main Content
    </div>
  </div>
</transition-stub>"
`;
